doc/source/f2py/code/CMakeLists_skbuild.txt

```diff
@@ -16,74 +15,81 @@ if(PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
-find_package(F2PY REQUIRED) # for ${F2PY_INCLUDE_DIR}
+
+# Grab the variables from a local Python installation
+# NumPy headers
+execute_process(
+  COMMAND "${PYTHON_EXECUTABLE}"
+  -c "import numpy; print(numpy.get_include())"
+  OUTPUT_VARIABLE NumPy_INCLUDE_DIRS
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+# F2PY headers
+execute_process(
+  COMMAND "${PYTHON_EXECUTABLE}"
+  -c "import numpy.f2py; print(numpy.f2py.get_include())"
+  OUTPUT_VARIABLE F2PY_INCLUDE_DIR
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
 
 # Prepping the module
 set(f2py_module_name "fibby")
 set(fortran_src_file "${CMAKE_SOURCE_DIR}/fib1.f")
-set(generated_module_file ${f2py_module_name}${PYTHON_EXTENSION_MODULE_SUFFIX})
+set(f2py_module_c "${f2py_module_name}module.c")
 
 # Target for enforcing dependencies
-add_custom_target(${f2py_module_name} ALL
+add_custom_target(genpyf
   DEPENDS "${fortran_src_file}"
-  )
-
-# Custom command for generating .c
+)
 add_custom_command(
-  OUTPUT "${f2py_module_name}module.c"
-  COMMAND ${F2PY_EXECUTABLE}
-    -m ${f2py_module_name}
-    ${fortran_src_file}
-    --lower
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS ${fortran_src_file}
-  )
+  OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${f2py_module_c}"
+  COMMAND ${PYTHON_EXECUTABLE}  -m "numpy.f2py"
+                   "${fortran_src_file}"
+                   -m "fibby"
+                   --lower # Important
+  DEPENDS fib1.f # Fortran source
+)
 
-add_library(${generated_module_file} MODULE
+add_library(${CMAKE_PROJECT_NAME} MODULE
             "${f2py_module_name}module.c"
             "${F2PY_INCLUDE_DIR}/fortranobject.c"
             "${fortran_src_file}")
 
-target_include_directories(${generated_module_file} PUBLIC
-                           ${F2PY_INCLUDE_DIRS}
+target_include_directories(${CMAKE_PROJECT_NAME} PUBLIC
+                           ${F2PY_INCLUDE_DIR}
+                           ${NumPy_INCLUDE_DIRS}
                            ${PYTHON_INCLUDE_DIRS})
-set_target_properties(${generated_module_file} PROPERTIES SUFFIX "")
-set_target_properties(${generated_module_file} PROPERTIES PREFIX "")
+set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES SUFFIX "${PYTHON_EXTENSION_MODULE_SUFFIX}")
+set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES PREFIX "")
 
 # Linker fixes
 if (UNIX)
   if (APPLE)
-    set_target_properties(${generated_module_file} PROPERTIES
+    set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES
     LINK_FLAGS  '-Wl,-dylib,-undefined,dynamic_lookup')
   else()
-    set_target_properties(${generated_module_file} PROPERTIES
+    set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES
   LINK_FLAGS  '-Wl,--allow-shlib-undefined')
   endif()
 endif()
 
-if (SKBUILD)
-  install(TARGETS ${generated_module_file} DESTINATION fibby)
-else()
-  install(TARGETS ${generated_module_file} DESTINATION ${CMAKE_SOURCE_DIR}/fibby)
-endif()
+add_dependencies(${CMAKE_PROJECT_NAME} genpyf)
+
+install(TARGETS ${CMAKE_PROJECT_NAME} DESTINATION fibby)
```

Comment:
I don't get why this disallows in-source builds, then puts the output targets, on an install no less, in source! Removed. If you want outputs in your source, do an in-source build. It doesn't somehow avoid the problems to do an out-of-source build then put the outputs back in the source, and it is all kinds of confusing.

Reply:
Fair enough. It was my understanding that there may be occasions where it makes sense to output only the extension module in the source and leave the rest in build but that seems rather insane when I think about it. I'd rather stick to the way this is now (disallowing in-source).

Reply:
The disallowing in-source build lines take up quite a bit of the examples, and are pretty unrelated to the examples themselves, and actually could get in the way (an in-source build is just like a `--inplace` build for setuptools, and those are/were common).

Reply:
I think it'd be a good idea to remove them from both in that case, but in a different `MAINT,DOC` PR (sorry).

There's always a little bit of a balancing act determining the minimum example without advancing bad practices.